ote_sdk/ote_sdk/entities/model_template.py

```diff
@@ -73,24 +73,132 @@ def __str__(self) -> str:
-    ROTATED_DETECTION = auto()
+    def __init__(
+        self,
+        value: int,
+        task_info: TaskInfo,
+    ):
+        """
+        :param value: Unique integer for .value property of Enum (auto() does not work)
+        :param task_info: NamedTuple containing information about the task's capabilities
+        """
+        self._value_ = value
+        self.domain = task_info.domain
+        self.is_trainable = task_info.is_trainable
+        self.is_anomaly = task_info.is_anomaly
+        self.is_global = task_info.is_global
+
+    NULL = 1, TaskInfo(
+        domain=Domain.NULL,
+        is_trainable=False,
+        is_anomaly=False,
+        is_global=False,
+        is_local=False,
+    )
+    DATASET = 2, TaskInfo(
+        domain=Domain.NULL,
+        is_trainable=False,
+        is_anomaly=False,
+        is_global=False,
+        is_local=False,
+    )
+    CLASSIFICATION = 3, TaskInfo(
+        domain=Domain.CLASSIFICATION,
+        is_trainable=True,
+        is_anomaly=False,
+        is_global=True,
+        is_local=False,
+    )
+    SEGMENTATION = 4, TaskInfo(
+        domain=Domain.SEGMENTATION,
+        is_trainable=True,
+        is_anomaly=False,
+        is_global=False,
+        is_local=True,
+    )
+    DETECTION = 5, TaskInfo(
+        domain=Domain.DETECTION,
+        is_trainable=True,
+        is_anomaly=False,
+        is_global=False,
+        is_local=True,
+    )
+    ANOMALY_DETECTION = 6, TaskInfo(
+        domain=Domain.ANOMALY_DETECTION,
+        is_trainable=True,
+        is_anomaly=True,
+        is_global=False,
+        is_local=True,
+    )
+    CROP = 7, TaskInfo(
+        domain=Domain.NULL,
+        is_trainable=False,
+        is_anomaly=False,
+        is_global=False,
+        is_local=False,
+    )
+    TILE = 8, TaskInfo(
+        domain=Domain.NULL,
+        is_trainable=False,
+        is_anomaly=False,
+        is_global=False,
+        is_local=False,
+    )
+    INSTANCE_SEGMENTATION = 9, TaskInfo(
+        domain=Domain.INSTANCE_SEGMENTATION,
+        is_trainable=True,
+        is_anomaly=False,
+        is_global=False,
+        is_local=True,
+    )
+    ACTIVELEARNING = 10, TaskInfo(
+        domain=Domain.NULL,
+        is_trainable=False,
+        is_anomaly=False,
+        is_global=False,
+        is_local=False,
+    )
+    ANOMALY_SEGMENTATION = 11, TaskInfo(
+        domain=Domain.ANOMALY_SEGMENTATION,
+        is_trainable=True,
+        is_anomaly=True,
+        is_global=False,
+        is_local=True,
+    )
+    ANOMALY_CLASSIFICATION = 12, TaskInfo(
+        domain=Domain.ANOMALY_CLASSIFICATION,
+        is_trainable=True,
+        is_anomaly=True,
+        is_global=True,
+        is_local=False,
+    )
+    ROTATED_DETECTION = 13, TaskInfo(
+        domain=Domain.ROTATED_DETECTION,
+        is_trainable=True,
+        is_anomaly=False,
+        is_global=False,
+        is_local=True,
+    )
 
     def __str__(self) -> str:
         return str(self.name)
```

Comment:
Can you replace it with 
```python
def __repr__(self) -> str:
    return self.name
```
It looks cleaner when you print a collection of `TaskType`, e.g. `print(list(TaskType))` or `[x for x in TaskType]`. The common usecase is when logging the list of supported types for a certain feature.